### DISCUSSION TITLE
Port setting in settings.yml doesn't work.

### DISCUSSION THREAD

**Discussion by aaqq0pp** [searxng/searxng#3764] [Q&A] [answered]
**Version of SearXNG, commit number if you are using on master branch and stipulate if you forked SearXNG**

  The latest build by the Installation Script

**How did you install SearXNG?**

  By the Installation Script

**What happened?**

   After finishing the installation, including the SearXNG uWSGI  instance, and everything was launched, the port listening didn't work! 

Background: I was using nginx stream to proxy TCP traffic at port 443 to backend servers (SearXNG is one of them), however I always got a nginx 111 unknown upstream error. I have no idea why it didn't work until I used `$ sudo ss -tuln | grep :<myport>` . I found out that the port is not listened.

**How To Reproduce**

 set a port in settings.yml, launch SearXNG uWSGI  instance server.
 use `$ sudo ss -tuln | grep :<myport>`  or other command.
the port is not listened.

**Expected behavior**

the port is listened.

**Screenshots & Logs**

**Additional context**

I'm a noob so if I made some stupid mistake please forgive me :(

**Comment by aaqq0pp** [comment, marked as answer]
Figured it out. uWSGI running nicely doesn't mean searxng is daemonised...I'm stupid.
Have to configure systemd to daemonise searxng.

**Reply by return42** [reply]
> After finishing the installation, including the SearXNG uWSGI instance, and everything was launched, the port listening didn't work!


If you have choose the defaults, uWSGI is listening on a unix socket.

**Reply by aaqq0pp** [reply]
> > After finishing the installation, including the SearXNG uWSGI instance, and everything was launched, the port listening didn't work!
> 
> If you have choose the defaults, uWSGI is listening on a unix socket.

Yes. And it is really confusing that the ip and port bonding setting only work when directly launch webapp.py via python. It should be explained at the basic setup page rather than hidden at server part.

**Reply by return42** [reply]
> I'm a noob so if I made some stupid mistake please forgive me :(

To understand some basic knowledge of http server and wsgi setup is needed .. I understand that is not trivial for a newbee .. but we can not cover all aspects of aspects of the setup .. socket or http upstream communication are just two alternative supported by the scripts when using nginx (or apache).

What the installations scripts does is described in the https://docs.searxng.org/admin/installation-nginx.html#nginx-s-searxng-site

![grafik](https://github.com/user-attachments/assets/31ae070e-8809-4f86-9644-bb967b04844f)

Socket is the default and the upstream communication is bound to `unix:: ..`

![grafik](https://github.com/user-attachments/assets/4b5a8061-308a-40da-9598-0c7d32d0545b)

**Reply by aaqq0pp** [reply]
> > I'm a noob so if I made some stupid mistake please forgive me :(
> 
> To understand some basic knowledge of http server and wsgi setup is needed .. I understand that is not trivial for a newbee .. but we can not cover all aspects of aspects of the setup .. socket or http upstream communication are just two alternative supported by the scripts when using nginx (or apache).
> 
> What the installations scripts does is described in the https://docs.searxng.org/admin/installation-nginx.html#nginx-s-searxng-site
> 
> ![grafik](https://private-user-images.githubusercontent.com/554536/361242251-31ae070e-8809-4f86-9644-bb967b04844f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2MDQ4NTQsIm5iZiI6MTcyNDYwNDU1NCwicGF0aCI6Ii81NTQ1MzYvMzYxMjQyMjUxLTMxYWUwNzBlLTg4MDktNGY4Ni05NjQ0LWJiOTY3YjA0ODQ0Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyNVQxNjQ5MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hYmUwZDlkODk4YTk2Nzc3MTY1YWE0NjY4MDM5YTQ1MDUxODY3MGIwNmI2YmUxNDBjOGJhZTdiNDY5Zjk5MWNkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NoWG4atfiu9phBVYthoEjVTa7CKxM37ssddvsLa66MQ)
> 
> Socket is the default and the upstream communication is bound to `unix:: ..`
> 
> ![grafik](https://private-user-images.githubusercontent.com/554536/361242327-4b5a8061-308a-40da-9598-0c7d32d0545b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ2MDQ4NTQsIm5iZiI6MTcyNDYwNDU1NCwicGF0aCI6Ii81NTQ1MzYvMzYxMjQyMzI3LTRiNWE4MDYxLTMwOGEtNDBkYS05NTk4LTBjN2QzMmQwNTQ1Yi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyNVQxNjQ5MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YmU0MDBmZDMwMDM4YjY1MGZkNDVhMTU0ZGFlZjJlYmFhODYwMWFiZDY3NDhiMjM1MTRmNmNhNDBhNzE5NzhkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.iUvDS7K_H1VmTwZYWeW-ofIakEvvZA-9m5LTR5wJPg0)

I see. It's all because of my stupidity anyway...Thank you for being kind and responsive :)
